packages/neptune-extension/src/common/components/notifications/notification/Notification.tsx

```diff
@@ -13,76 +13,52 @@ import { UpgradeAvailableNotification } from 'common/components/notifications/re
 
 
 // Module
-type NotificationProps = NotificationPayload & {
-  removeNotification: (id: string) => void
-};
-
-class Notification extends React.PureComponent<NotificationProps> {
-  SUCCEED_TIMEOUT_TIME = 5000;
-
-  timer: number | undefined = undefined;
-
-  componentDidMount() {
-    this.setTimer();
-  }
-
-  componentWillUnmount() {
-    this.clearTimer();
-  }
-
-  clearTimer = () => {
-    window.clearTimeout(this.timer);
-  }
-
-  setTimer = () => {
-    this.timer = window.setTimeout(this.remove, this.SUCCEED_TIMEOUT_TIME);
-  }
-
-  remove = () => {
-    const {
-      id,
-      removeNotification,
-    } = this.props;
-
-    removeNotification(id);
-  }
-
-  render() {
-    switch (this.props.type) {
-      case 'checkpoint-successful': {
-        return (
-          <CheckpointSuccessfulNotification
-            type={this.props.type}
-            data={this.props.data}
-            onClose={this.remove}
-          />
-        );
-      }
+const SUCCEED_TIMEOUT_TIME = 5000;
+const Notification: React.FC<NotificationPayload> = (props) => {
+  const dispatch = useDispatch();
+  const remove = React.useCallback(
+    () => dispatch(removeNotification(props.id)),
+    [props.id]
+  );
+
+  React.useEffect(() => {
```

Comment:
How about stopping the timer on mouse enter and starting it again on mouse leave?

Reply:
I don't see such feature in previous implementation.